### PR TITLE
Add documentation for plugins using generic models to get change logging

### PR DESCRIPTION
**Pull request by glennmatthews**
### Fixes: #15

No actual code changes are required for plugins to be able to use ChangeLoggedModel, but in most cases a plugin that wants change-logging is better off using one of the `generics` models rather than mixing in ChangeLoggedModel directly. I've added a comparison of model options to the plugin development docs and made a few related corrections that I noticed along the way.

### CODE REVIEW

**Review by jathanism — APPROVED** [review]
No complaints! Thank you for doing this.

File: nautobot/docs/plugins/development.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ All required settings must be configured by the user. If a configuration paramet
 
 ### Install the Plugin for Development
 
-The plugin needs to be installed into the same python environment where Nautobot is, so that we can get access to `nautobot-server` command, and also so that the nautobot-server is aware of the new plugin.
```
Comment (jvanderaa) [inline nautobot/docs/plugins/development.md:159]: This double space was intentional for the new paragraph.

Comment (glennmatthews) [reply nautobot/docs/plugins/development.md:159]: Not sure I understand your comment, sorry!

Comment (jvanderaa) [reply nautobot/docs/plugins/development.md:159]: The double space that was being removed on this line was intended to make a new paragraph (or a line break) - https://meta.stackoverflow.com/editing-help

Comment (glennmatthews) [reply nautobot/docs/plugins/development.md:159]: We already get a paragraph break from the blank line at line 160 - here's how this PR is rendering on RTD:

image

**Pull request merged**